compiler/noirc_frontend/src/elaborator/mod.rs

```diff
@@ -1297,6 +1297,16 @@ impl<'context> Elaborator<'context> {
         self.current_item = Some(DependencyId::Global(global_id));
         let let_stmt = global.stmt_def;
 
+        let name = if self.is_in_lsp_mode() {
+            if let Pattern::Identifier(ident) = &let_stmt.pattern {
+                Some(ident.to_string())
+            } else {
+                None
+            }
+        } else {
+            None
+        };
```

Comment:
Can you move this to a method on `Pattern`? You can probably reuse name_ident since we don't allow tuple globals.